presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1594,9 +1594,17 @@ private Scope computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope ou
                     .collect(toImmutableList());
 
             // generate placeholder fields
+            Set<Field> seen = new HashSet<>();
```

Comment:
In https://github.com/prestodb/presto/pull/8252/files#diff-1a785e48f45c659076f1e4ec089517deR1590 I don't think it needed to filter aggregates which are just pure column references.